src/scenic/syntax/compiler.py

```diff
@@ -4,42 +4,162 @@
     node = ast.fix_missing_locations(compiler.visit(scenicAST))
     return node, compiler.requirements
 
+
 # shorthands for convenience
 
 loadCtx = ast.Load()
 
 # transformer
 
+
 class ScenicToPythonTransformer(ast.NodeTransformer):
     def __init__(self) -> None:
         super().__init__()
         self.requirements = []
-    
+
     def generic_visit(self, node):
         if isinstance(node, s.AST):
-            raise Exception(f'Scenic AST node "{node.__class__.__name__}" needs visitor in compiler')
+            raise Exception(
+                f'Scenic AST node "{node.__class__.__name__}" needs visitor in compiler'
+            )
         return super().generic_visit(node)
-    
+
     # Special Case
 
     def visit_EgoAssign(self, node: s.EgoAssign):
-        return ast.Expr(value=ast.Call(
-            func=ast.Name(id="ego", ctx=loadCtx),
-            args=[self.visit(node.value)],
-            keywords=[],
-        ))
+        return ast.Expr(
+            value=ast.Call(
+                func=ast.Name(id="ego", ctx=loadCtx),
+                args=[self.visit(node.value)],
+                keywords=[],
+            )
+        )
 
     # Instance & Specifier
 
     def visit_New(self, node: s.New):
         return ast.Call(
             func=ast.Name(id=node.className, ctx=loadCtx),
+            args=[self.visit(s) for s in node.specifiers],
+            keywords=[],
+        )
+
+    def visit_WithSpecifier(self, node: s.WithSpecifier):
+        return ast.Call(
+            func=ast.Name(id="With", ctx=loadCtx),
+            args=[
+                ast.Constant(value=node.prop),
+                self.visit(node.value),
+            ],
+            keywords=[],
+        )
+
+    def visit_AtSpecifier(self, node: s.AtSpecifier):
+        return ast.Call(
+            func=ast.Name(id="At", ctx=loadCtx),
+            args=[self.visit(node.position)],
+            keywords=[],
+        )
+
+    def visit_OffsetBySpecifier(self, node: s.OffsetBySpecifier):
+        return ast.Call(
+            func=ast.Name(id="OffsetBy", ctx=loadCtx),
+            args=[
+                self.visit(node.offset),
+            ],
+            keywords=[],
+        )
+
+    def visit_OffsetAlongSpecifier(self, node: s.OffsetAlongSpecifier):
+        return ast.Call(
+            func=ast.Name(id="OffsetAlongSpec", ctx=loadCtx),
+            args=[
+                self.visit(node.direction),
+                self.visit(node.offset),
+            ],
+            keywords=[],
+        )
+
+    def visit_PositionSpecifier(self, node: s.PositionSpecifier):
+        if isinstance(node.direction, s.LeftOf):
+            fn = "LeftSpec"
+        elif isinstance(node.direction, s.RightOf):
+            fn = "RightSpec"
+        elif isinstance(node.direction, s.AheadOf):
+            fn = "Ahead"
+        elif isinstance(node.direction, s.Behind):
+            fn = "Behind"
+        else:
+            raise TypeError(
```

Comment:
This should be unreachable unless there's a bug in the parser, right? So it should be an `assert` or something else indicating an internal error.

Reply:
Changed to assert
8713ad19a3c3f7d2ab58aa4bb60419c95502d797